networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/rpc/beaconchain/BeaconChainMethods.java

```diff
@@ -25,57 +28,126 @@
+    return new BeaconChainMethods(
+        createStatus(statusMessageFactory, peerLookup),
+        createGoodBye(metricsSystem, peerLookup),
+        createBeaconBlocksByRoot(chainStorageClient, peerLookup),
+        createBeaconBlocksByRange(combinedChainDataClient, peerLookup));
+  }
+
+  private static Eth2RpcMethod<StatusMessage, StatusMessage> createStatus(
+      final StatusMessageFactory statusMessageFactory, final PeerLookup peerLookup) {
     final StatusMessageHandler statusHandler = new StatusMessageHandler(statusMessageFactory);
+    return new Eth2RpcMethod<>(
+        STATUS,
+        RpcEncoding.SSZ,
+        StatusMessage.class,
+        StatusMessage.class,
+        false,
+        statusHandler,
+        peerLookup);
+  }
+
+  private static Eth2RpcMethod<GoodbyeMessage, GoodbyeMessage> createGoodBye(
+      final MetricsSystem metricsSystem, final PeerLookup peerLookup) {
     final GoodbyeMessageHandler goodbyeHandler = new GoodbyeMessageHandler(metricsSystem);
+    return new Eth2RpcMethod<>(
+        GOODBYE,
+        RpcEncoding.SSZ,
+        GoodbyeMessage.class,
+        GoodbyeMessage.class,
+        true,
+        goodbyeHandler,
+        peerLookup);
+  }
+
+  private static Eth2RpcMethod<BeaconBlocksByRootRequestMessage, BeaconBlock>
+      createBeaconBlocksByRoot(
+          final ChainStorageClient chainStorageClient, final PeerLookup peerLookup) {
     final BeaconBlocksByRootMessageHandler beaconBlocksByRootHandler =
         new BeaconBlocksByRootMessageHandler(chainStorageClient);
+    return new Eth2RpcMethod<>(
+        BEACON_BLOCKS_BY_ROOT,
+        RpcEncoding.SSZ,
+        BeaconBlocksByRootRequestMessage.class,
+        BeaconBlock.class,
+        false,
+        beaconBlocksByRootHandler,
+        peerLookup);
+  }
+
+  private static Eth2RpcMethod<BeaconBlocksByRangeRequestMessage, BeaconBlock>
+      createBeaconBlocksByRange(
+          final CombinedChainDataClient combinedChainDataClient, final PeerLookup peerLookup) {
+
     final BeaconBlocksByRangeMessageHandler beaconBlocksByRangeHandler =
         new BeaconBlocksByRangeMessageHandler(combinedChainDataClient);
-    return new RpcMethods(
-        new RpcMessageHandler<>(STATUS, peerLookup, statusHandler),
-        new RpcMessageHandler<>(GOODBYE, peerLookup, goodbyeHandler).setCloseNotification(),
-        new RpcMessageHandler<>(BEACON_BLOCKS_BY_ROOT, peerLookup, beaconBlocksByRootHandler),
-        new RpcMessageHandler<>(BEACON_BLOCKS_BY_RANGE, peerLookup, beaconBlocksByRangeHandler));
+    return new Eth2RpcMethod<>(
+        BEACON_BLOCKS_BY_RANGE,
+        RpcEncoding.SSZ,
+        BeaconBlocksByRangeRequestMessage.class,
+        BeaconBlock.class,
+        false,
+        beaconBlocksByRangeHandler,
+        peerLookup);
+  }
+
+  public Collection<RpcMethod> all() {
+    return Collections.unmodifiableCollection(allMethods);
+  }
+
+  public Eth2RpcMethod<StatusMessage, StatusMessage> status() {
```

Comment:
Strictly these should probably be `get` prefixed, but it is one of those cases where it feels right to break the rule (and I'm usually a big fan of `get` prefixes...).